src/features/helpers/bignumber.js

```diff
@@ -85,6 +85,11 @@ export function subtract(numberOne, numberTwo) {
   return new BigNumber(`${numberOne}`).minus(new BigNumber(`${numberTwo}`)).toString();
 }
 
+export function convertBalanceToDecimals(value, decimals = 18) {
+  if (decimals >= 18) return value;
+  return multiply(value, 10 ** (18 - decimals));
+}
+
```

Comment:
I would suggest to avoid ambiguous logic, all token balances should be processed in the same way. Please try to find the root cause, probably somewhere decimals are hardcoded. Once fixed, usage of `byDecimals` should be enough to show correct balances.